clayui.com/plugins/gatsby-plugin-svg-compile-sprite/gatsby-node.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * SPDX-FileCopyrightText: © 2018 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+// Compiles SVG icons in directory `clay-css/src/images/icons` into a spritemap
+
+const fs = require('fs');
+const path = require('path');
+
+const generateSvgSprite = (pluginOptions) => {
+	const REGEX_FILE_EXT_SVG = /(flags-|.svg$)/g;
+	const REGEX_HTML_COMMENTS = /<!--(.+?)(-->\n|-->)/gs;
+	const REGEX_SVG_VIEWBOX = /(\sv|v)iewBox="(.+?)"/g;
+	const REGEX_SVG_TAG = /(<\/svg|<svg(.+?))>(\n|)/g;
+
+	const filesArr = fs.readdirSync(pluginOptions.srcDir);
+
+	let strSprite = `<?xml version="1.0" encoding="UTF-8"?><!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd"><svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">`;
+
+	filesArr.forEach((file) => {
+		let icon = fs
+			.readFileSync(path.join(pluginOptions.srcDir, file), 'utf8')
+			.toString();
+
+		const id = file.replace(REGEX_FILE_EXT_SVG, '').toLowerCase();
+
+		const viewBox = icon.match(REGEX_SVG_VIEWBOX);
+
+		const symbol = `<symbol id="${id}"${viewBox}>`;
```

Comment:
If this ever failed to match, it would use `null` as the `viewBox`. Maybe an `if` and a `throw` is warranted?

Reply:
Thanks @wincent! Your reviews are always appreciated. I learn more every time. I'll make all these changes.